subgraphs/uniswap-v2/src/common/tokens.ts

```diff
@@ -1,87 +1,120 @@
 /* eslint-disable prefer-const */
 import { ERC20 } from '../../generated/Factory/ERC20'
-import { ERC20SymbolBytes } from '../../generated/Factory/ERC20SymbolBytes'
-import { ERC20NameBytes } from '../../generated/Factory/ERC20NameBytes'
-import { StaticTokenDefinition } from './staticTokenDefinition'
 import { Address } from '@graphprotocol/graph-ts'
-import { ERROR_NUM } from '../common/constants'
+import { DEFAULT_DECIMALS } from '../common/constants'
+import { Token } from '../../generated/schema'
 
-
-export function fetchTokenSymbol(tokenAddress: Address): string {
-  let contract = ERC20.bind(tokenAddress)
-  let contractSymbolBytes = ERC20SymbolBytes.bind(tokenAddress)
-
-  // try types string and bytes32 for symbol
-  let symbolValue = 'unknown'
-  let symbolResult = contract.try_symbol()
-  if (symbolResult.reverted) {
-    let symbolResultBytes = contractSymbolBytes.try_symbol()
-    if (!symbolResultBytes.reverted) {
-      // for broken pairs that have no symbol function exposed
-      if (!isNullEthValue(symbolResultBytes.value.toHexString())) {
-        symbolValue = symbolResultBytes.value.toString()
-      } else {
-        // try with the static definition
-        let staticTokenDefinition = StaticTokenDefinition.fromAddress(tokenAddress)
-        if(staticTokenDefinition != null) {
-          symbolValue = staticTokenDefinition.symbol
-        }
-      }
-    }
-  } else {
-    symbolValue = symbolResult.value
+export function getOrCreateToken(address: Address): Token {
+  let id = address.toHexString();
+  let token = Token.load(id);
+  if (!token) {
+    token = new Token(id);
+    let erc20Contract = ERC20.bind(address);
+    let decimals = erc20Contract.try_decimals();
+    // Using try_cause some values might be missing
+    let name = erc20Contract.try_name();
+    let symbol = erc20Contract.try_symbol();
+    // TODO: add overrides for name and symbol
+    token.decimals = decimals.reverted ? DEFAULT_DECIMALS : decimals.value;
+    token.name = name.reverted ? '' : name.value;
+    token.symbol = symbol.reverted ? '' : symbol.value;
+    token.save();
   }
+  return token as Token;
+}
 
-  return symbolValue
+export function getOrCreateLPToken(tokenAddress: Address, token0: Token, token1: Token): Token {
+  let id = tokenAddress.toHexString();
+  let token = Token.load(id)
+  // fetch info if null
+  if (token === null) {
+      token = new Token(tokenAddress.toHexString())
+      token.symbol = token0.name + '/' + token1.name
+      token.name = token0.name + '/' + token1.name + " LP"
+      token.decimals = DEFAULT_DECIMALS
+      token.save()
+  }
+  return token
 }
 
-export function fetchTokenName(tokenAddress: Address): string {
-  let contract = ERC20.bind(tokenAddress)
-  let contractNameBytes = ERC20NameBytes.bind(tokenAddress)
+// export function fetchTokenSymbol(tokenAddress: Address): string {
```

Comment:
why are these commented out? Should either uncomment or remove

Reply:
Removed